lib/dry/validation/contract/class_interface.rb

```diff
@@ -17,7 +17,8 @@ module ClassInterface
         #
         # @api public
         def params(&block)
-          @__schema__ ||= Schema.Params(schema_opts, &block)
+          raise ::Dry::Validation::DuplicateSchemaError, 'Schema has alredy been defined' if defined?(@__schema__)
+          @__schema__ = Schema.Params(schema_opts, &block)
```

Comment:
it'd be good to avoid the duplication and introduce a common method, something like `define(:Params, &block)` and handle this check / raising there.